include/graaflib/algorithm/cycle_detection.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <graaflib/graph.h>
+#include <graaflib/types.h>
+
+#include <concepts>
+
+namespace graaf::algorithm {
+
+/*
+ * @breif Traverses the graph and check for cycles.
+ *
+ * @param graph The graph to traverse.
+ */
+template <typename V, typename E, graph_spec S>
+bool has_cycle(const graph<V, E, S> &graph);
```

Comment:
[very minor]: I would prefer to mark predicates such as these with `[[nodiscard]]`

Reply:
Ok